File: Greedy/763. Partition Labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        vector<int> ans;
        int n = size(s);
        int start = 0;
        int end = 0;
        unordered_map<char, int> d;
        for (int i = 0; i < n; i++){
            d[s[i]] = i;
        }
        for (int i = 0; i < n; i++){
            end = max(end, d[s[i]]);
            if (i == end){
                ans.push_back(end - start + 1);
                start = end + 1;
            }
        }
        return ans;
    }
};

File: Greedy/763. Partition Labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        d = {}
        for i, ch in enumerate(s):
            d[ch] = i
        start = 0
        end = 0
        ans = []
        for i, ch in enumerate(s):
            end = max(end, d[ch])
            if i == end:
                ans.append(end - start + 1)
                start = end + 1
        return ans